app-web/src/components/Form/TopicForm.js

```diff
@@ -140,31 +177,34 @@ export const TopicForm = () => {
   );
 };
 
-const convertToRegistryFormat = values => {
+const convertToRegistryFormat = (values, operation) => {
   const convertedFields = {
     name: values.topicName,
     description: values.topicDescription,
     sourceProperties: {
       sources: values.sources.map(source => ({
         sourceType: source.sourceType,
-        sourceProperties: getSourceProps(source),
+        sourceProperties: getSourceProps(source, operation),
         resourceType: source.resourceType,
       })),
     },
   };
   return convertedFields;
 };
 
-const getSourceProps = source => {
+const getSourceProps = (source, operation) => {
   const properties = {};
-  properties.url = source.url;
+  properties.url = source.sourceProperties.url;
   if (source.sourceType === 'github') {
-    properties.owner = source.owner;
-    properties.repo = source.repo;
-    properties.files = [source.file];
+    properties.owner = source.sourceProperties.owner;
+    properties.repo = source.sourceProperties.repo;
+    properties.files =
+      operation === 'create'
```

Comment:
can you explain whats going on here?

Reply:
There is a slight change only with the **files** input field. When creating a topic, the input field works as usual, however when editing a topic, the files  data comes in as an array. So there needs to be converted to a string before any other operations can be done.